core/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -966,10 +966,20 @@ public void testTranslogWriter() throws IOException {
         final int numOps = randomIntBetween(10, 100);
         byte[] bytes = new byte[4];
         ByteArrayDataOutput out = new ByteArrayDataOutput(bytes);
+        long minSeqNo = Long.MAX_VALUE;
+        long maxSeqNo = Long.MIN_VALUE;
+        final Set<Long> seenSeqNos = new HashSet<>();
         for (int i = 0; i < numOps; i++) {
             out.reset(bytes);
             out.writeInt(i);
-            writer.add(new BytesArray(bytes));
+            long seqNo;
+            do {
```

Comment:
once we add it, can we test that we do the right things with unassigned seq nos due to BWC?